src/core/server/saved_objects/import/utils.ts

```diff
@@ -91,6 +91,25 @@ export const updateDataSourceNameInVegaSpec = (
       });
 };
 
+export const updateDataSourceNameInTimeline = (
+  timelineExpression: string,
+  dataSourceTitle: string
+) => {
+  const expressionRegex = /\.(opensearch|es|elasticsearch)\(([^)]*)\)/g;
+
+  const replaceCallback = (match: string, funcName: string, args: string) => {
+    if (!args.includes('data_source_name')) {
+      args = args.trim();
+      args = `${args}, data_source_name=${dataSourceTitle}`;
```

Comment:
We should wrap the `dataSourceTitle` in `""` since titles can have spaces (ex: `dataSourceTitle="Some DataSource A"`)

```suggestion
      args = `${args}, data_source_name="${dataSourceTitle}"`;
```

Reply:
And update relevant test cases as well